boxes.cc

```diff
@@ -109,28 +106,17 @@ void BoxesTest::Boxes(const std::string &_physicsEngine
     E0 = 368.54641249999997;
   }
 
-  for (int i = 0; i < _modelCount; ++i)
-  {
-    // give models unique names
-    msgModel.set_name(this->GetUniqueString("model"));
-    // give models unique positions
-    msgs::Set(msgModel.mutable_pose()->mutable_position(),
-              ignition::math::Vector3d(0.0, dz*2*i, 0.0));
-
-    model = this->SpawnModel(msgModel);
-    ASSERT_NE(model, nullptr);
-
+  for (auto model : models) {
     link = model->GetLink();
     ASSERT_NE(link, nullptr);
-
-    // Set initial conditions
-    link->SetLinearVel(v0);
-    link->SetAngularVel(w0);
   }
+  // adding a small delay (waiting for the model to load properly)
+  common::Time::MSleep(50);
+
   ASSERT_EQ(v0, link->WorldCoGLinearVel());
```

Comment:
it could be nice to also check that the moment of inertia matrix in the body-fixed frame matches `I0`

Reply:
Hi Steve, I have made the suggested changes could you please review them again ?